core/telemetry.go

```diff
@@ -57,13 +83,19 @@ func AroundFunc(ctx context.Context, self dagql.Object, id *call.ID) (context.Co
 		defer telemetry.End(span, func() error { return err })
 
 		if cached {
-			// TODO maybe this should be an event?
+			// NOTE: this is never actually called on cache hits, but might be in the
+			// future.
 			span.SetAttributes(attribute.Bool(telemetry.CachedAttr, true))
 		}
 
+		if ctx.Err() != nil {
+			// If the request was canceled, reflect it on the span.
+			span.SetAttributes(attribute.Bool(telemetry.CanceledAttr, true))
+		}
```

Comment:
Don't think we need to; the only possible `ctx.Err()` is rooted in either `context.Canceled` or `context.DeadlineExceeded`, and we don't make a distinction for this status at the moment.

I suppose we could add a `TimeoutAttr` sometime but I'm not sure if it would even realistically be detected in all cases; from the this code's POV a timeout enforced client-side will always just look like a canceled request.